code/game/supplyshuttle.dm

```diff
@@ -147,6 +148,10 @@ var/datum/controller/supply/supply_controller = new()
 
 	//If any computers are able to order contraband, it's enabled. Otherwise, it's disabled!
 
+/obj/structure/machinery/computer/supplycomp/proc/lock_black_market(market_locked = FALSE)
+	if(market_locked)
+		black_market_lockout = TRUE
```

Comment:
I'm pretty sure this proc is supposed to black market lockout all the currently connected computers. The issue with it before was not the for loop but that the for loop wasn't actually interacting with the bound computers it is meant to be locking but instead just resetting the one computer's lockout value.

Reply:
Oh and document the proc please (if I'm correct)